packages/backend-defaults/src/entrypoints/httpAuth/httpAuthServiceFactory.ts

```diff
@@ -71,23 +70,54 @@ type RequestWithCredentials = Request & {
   [limitedCredentialsSymbol]?: Promise<BackstageCredentials>;
 };
 
-class DefaultHttpAuthService implements HttpAuthService {
+/**
+ * @public
+ * Options for creating a DefaultHttpAuthService.
+ */
+export interface DefaultHttpAuthServiceOptions {
+  auth: AuthService;
+  discovery: DiscoveryService;
+  pluginId: string;
+  // Optionally override the default token extraction logic
```

Comment:
Might as well make this a `/** */` doc while at it 😁 

Could you also document this new customizability at https://github.com/backstage/backstage/blob/master/docs/backend-system/core-services/http-auth.md#configuring-the-service?